gammapy/cube/models.py

```diff
@@ -58,13 +58,29 @@ class SkyModels(SkyModelBase):
     __slots__ = ["skymodels"]
 
     def __init__(self, skymodels):
+
+        existing_names = []
+
+        for model in skymodels:
+            if model.name in existing_names:
+                raise ValueError("SkyModel '{}' already exists, please choose"
+                                 " another name.".format(model.name))
+            existing_names.append(model.name)
+
         self.skymodels = skymodels
-        parameters = []
 
-        for skymodel in skymodels:
+    @property
+    def parameters(self):
+        parameters = []
+        for skymodel in self.skymodels:
             for p in skymodel.parameters:
                 parameters.append(p)
-        super().__init__(parameters)
+        return Parameters(parameters)
```

Comment:
Yes, I saw that step by step the `SkyModels` class evolved from a `Model` to more something like a model container (contains multiple models, supports I/O, access model by name, etc.). So I thought it might be better to remove the inheritance from the Model base class...we don't loose any functionality from by doing this, but the class structure is maybe a little bit cleaner.